bot/exts/fun/quack.py

```diff
@@ -0,0 +1,80 @@
+import random
+from typing import Literal, Optional
+
+import discord
+from aiohttp import client_exceptions
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Colours, NEGATIVE_REPLIES
+
+API_URL = 'https://quackstack.pythondiscord.com'
+
+
+class Quackstack(commands.Cog):
+    """Cog used for wrapping Quackstack."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @commands.command()
+    async def quack(
+        self,
+        ctx: commands.Context,
+        ducktype: Literal["duck", "manduck"] = "duck",
+        *,
+        seed: Optional[str] = None
+    ) -> None:
+        """
+        Use the Quackstack API to generate a random duck.
+
+        If a seed is provided, a duck is generated based on the given seed.
+        Either "duck" or "manduck" can be provided to change the duck type generated.
+        """
+        error_embed = discord.Embed(
+            title=random.choice(NEGATIVE_REPLIES),
+            description="The request failed. Please try again later.",
+            color=Colours.soft_red,
+        )
+
+        ducktype = ducktype.lower()
+        if ducktype not in ["duck", "manduck"]:
+            raise commands.BadArgument(
+                "The duck type can only be `duck` or `manduck`."
+            )
+
+        quackstack_url = f"{API_URL}/{ducktype}"
+        params = {}
+        if seed is not None:
+            try:
+                seed = int(seed)
+            except ValueError:
+                # Converts any word(s) with numbers into an integer
```

Comment:
```suggestion
                # We just need to turn the string into an integer any way possible
```
May sound better like this? Feel free to keep the old comment though.